sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/UnsafeRowConverter.scala

```diff
@@ -72,6 +70,19 @@ class UnsafeRowConverter(fieldTypes: Array[DataType]) {
    */
   def writeRow(row: InternalRow, baseObject: Object, baseOffset: Long): Int = {
     unsafeRow.pointTo(baseObject, baseOffset, writers.length, null)
+
+    if (writers.length > 0) {
```

Comment:
I think that we need to change this to `>=` then.  If you look at the line below, on line 77, when `writers.length` == 0 then `n == 0` but the loop implies that we'll still allocate one word for null-tracking bits.